go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -451,6 +451,9 @@ func (c *Controller) getConfigAndUpdateRefs(eIP *eipv1.EgressIP, updateRefs bool
 		}
 		c.referencedObjects[eIP.Name] = refObjs
 	}
+	if eIPConfig == nil || podIPConfigs == nil {
```

Comment:
NIT: with the changes in `processEIP` it should be enough to just check one of these 2, but I guess you are taking no chances here. :)